packages/theme/src/cli/utilities/asset-ignore.test.ts

```diff
@@ -171,6 +171,7 @@ describe('asset-ignore', () => {
 
       // Then
       expect(actualChecksums).toEqual([
+        {key: 'templates/404.json', checksum: '6666666666666666666666666666666'},
```

Comment:
Actually, I don't think this is what we expect, as when we run:
```bash
shopify theme push -u -t "theme-test-1" --ignore "templates/**/*.json" --legacy --verbose
```
We get these logs:
```bash
┃ DEBUG ignore templates/customers/account.json
┃ DEBUG ignore templates/customers/activate_account.json
┃ DEBUG ignore templates/customers/addresses.json
┃ DEBUG ignore templates/customers/login.json
┃ DEBUG ignore templates/customers/order.json
┃ DEBUG ignore templates/customers/register.json
┃ DEBUG ignore templates/customers/reset_password.json
```

So, this would not a backward-compatible change. The https://github.com/Shopify/cli/pull/4564 PR intentionally introduces this change to make the ignore filter backward-compatible with the former Ruby implementation based on `File.fnmatch?`:

![image](https://github.com/user-attachments/assets/676e10c3-e49f-439c-b58e-45a1791e8e3e)